Sources/MapboxDirections/Directions.swift

```diff
@@ -375,6 +384,101 @@ open class Directions: NSObject {
         return requestTask
     }
     
+    @discardableResult open func refresh(route: Route, currentLegIndex: Int, completionHandler: @escaping RouteRefreshCompletionHandler) -> URLSessionDataTask? {
+        
+        guard let routeIdentifier = route.routeIdentifier, let routeIndex = route.routeIndex else {
```

Comment:
Instead of persisting the original index as part of the route, we can have this method take the whole RouteResponse and a route index. The original goal of #391 was that the navigation SDK would keep track of the entire RouteResponse and a route index, or the route and route index. That would require some straightforward but fairly invasive refactoring of the public API. I think we need to make that change ahead of v1.1 anyhow, because this is only the first of several features that will require context about more than the individual route that the user selected.

Reply:
To keep this refactor from ballooning in scope at the tail end of v1.0, we’re going to avoid pulling in the whole RouteResponse for now. In fact, I’ve refactored this method to take the response identifier itself, to make it easier to remove the `Route.routeIdentifier` property in #452.